packages/adblocker/src/engine/engine.ts

```diff
@@ -697,23 +766,46 @@ export default class FilterEngine extends EventEmitter<
     url,
     hostname,
     domain,
+
+    reference,
   }: {
     url: string;
     hostname: string;
     domain: string | null | undefined;
+
+    reference?: any | undefined;
```

Comment:
if it's optional do you need to specify `undefined`?

Reply:
Just for the comfort. It ensures we can completely put an argument optional in case of given "undefined" value. The argument is initialised regardless of the value existence.